pkg/controller/daemon/daemon_controller.go

```diff
@@ -684,12 +684,8 @@ func nodeInSameCondition(old []v1.NodeCondition, cur []v1.NodeCondition) bool {
 }
 
 func shouldIgnoreNodeUpdate(oldNode, curNode v1.Node) bool {
-	if !nodeInSameCondition(oldNode.Status.Conditions, curNode.Status.Conditions) {
-		return false
-	}
-	oldNode.ResourceVersion = curNode.ResourceVersion
-	oldNode.Status.Conditions = curNode.Status.Conditions
-	return apiequality.Semantic.DeepEqual(oldNode, curNode)
+	return apiequality.Semantic.DeepEqual(oldNode.GetLabels(), curNode.GetLabels()) &&
+		apiequality.Semantic.DeepEqual(oldNode.Spec.Taints, curNode.Spec.Taints)
```

Comment:
if any other mutable properties are used in NodeShouldRunDaemonPod or anything it calls (like `k8s.io/component-helpers/scheduling/corev1/nodeaffinity#extractNodeFields`), this will need updating.

Worth adding a comment here and in those functions mentioning that

Reply:
@liggitt Thanks for the code review, shouldIgnoreNodeUpdate and the NodeShouldRunDaemonPod function mentioned in comment have been added, please take a look.